### DISCUSSION TITLE
replacing outfit on the fly

### DISCUSSION THREAD

**Discussion by branaway** [pixiv/three-vrm#1098] [Help]
I'm exploring how to replace the texture image of the outfit for an example, effectively replacing the look of the outfit on the fly. Any idea?

**Comment by branaway** [comment]
Replacing the hair hierarchy is another thing I'm working on...

**Comment by tejasnayak25** [comment]
I'm exploring the same. The method to change outfits and control wind will be super helpful. 
Please help

**Comment by 0b5vr** [comment]
Replacing outfits is currently something both VRM and three-vrm do not cover as a feature.

Needs more context. What kind of model and outfit do you have right now?
I currently believe such functions should be addressed on the application side

**Reply by tejasnayak25** [reply]
I have a male model with a school uniform. I want it to switch between the uniform and other outfits when required. 
Any ideas?
And how do I create wind effect in three-vrm?
Please help

**Reply by branaway** [reply]
as long as the VRM model has a common naming standards for different part of the clothing mesh and texture, we can have something like:  myAvatar.shirt.texture = myNewTxture to change the shirt texture, and myAvatar.outfit.remove() to remove the outfit of the avatar. 

From what I have checked on the VRM models in the market, there is no standard conventions of naming clothing on an avatar.  I really hope that vroid studio is open sourced so the community can play with it and come up with something that separate the clothing from the model body. There are lots of use cases in games and entertainment.

**Reply by tejasnayak25** [reply]
Oh, I understand. But how can I use wind effect? Kindly let me know. Thank you

**Reply by tejasnayak25** [reply]
From what I learnt from the above mentioned comment, there's this method `myAvatar.outfit.remove();`
Can't we just load another model and do this,
`myAvatar.outfit.add(myAvatar2.outfit);`

Or something like 
`myAvatar.outfit = myAvatar2.outfit;`

Is there an add method available in three-vrm?

**Comment by tejasnayak25** [comment]
I have a male model with a school uniform. I want it to switch between the uniform and other outfits when required.
Any ideas?
And how do I create wind effect in three-vrm?
Please help

**Reply by 0b5vr** [reply]
That's unrelated to the discussion itself. Please ask in a separate thread

**Comment by tejasnayak25** [comment]
There is another solution for this.

You have to load two models, i.e., one with body texture and other with only cloth texture (Erase the body texture).
<pre>
<code>
loader1.load('char.vrm', (gltf) => {
let vrm = gltf.userData.vrm;
scene.add(vrm.scene);
}
loader2.load('cloth.vrm', (gltf) => {
let vrm = gltf.userData.vrm;
scene.add(vrm.scene);
}
</code>
</pre>
This will add two models to the scene which by default will overlap eachother.
Further if you want to make changes like rotating the model or moving it, you just have make changes for both,i.e,
<pre>
model1.rotation.set(1,2,1);
model2.rotation.set(1,2,1);
</pre>
This is one of the solutions for your problem I believe. Let me know if it helped.
